lib/ansible/module_utils/nxos.py

```diff
@@ -149,55 +144,41 @@ def run_commands(self, commands, check_rc=True):
             else:
                 cmd = item['command']
 
-            rc, out, err = self.exec_command(cmd)
+            out = connection.get(cmd)
             try:
                 out = to_text(out, errors='surrogate_or_strict')
             except UnicodeError:
                 self._module.fail_json(msg=u'Failed to decode output from %s: %s' % (cmd, to_text(out)))
 
-            if check_rc and rc != 0:
-                self._module.fail_json(msg=to_text(err))
-
-            if not check_rc and rc != 0:
-                try:
-                    out = self._module.from_json(err)
-                except ValueError:
-                    out = to_text(err).strip()
-            else:
-                try:
-                    out = self._module.from_json(out)
-                except ValueError:
-                    out = to_text(out).strip()
-
-            if item['output'] == 'json' and out != '' and isinstance(out, string_types):
-                self._module.fail_json(msg='failed to retrieve output of %s in json format' % item['command'])
+            try:
+                out = self._module.from_json(out)
+            except ValueError:
+                out = to_text(out).strip()
 
             responses.append(out)
         return responses
 
     def load_config(self, config, return_error=False, opts=None):
         """Sends configuration commands to the remote device
         """
-        if opts is None:
-            opts = {}
+        connection = self._get_connection()
+        connection.edit_config(config)
 
-        rc, out, err = self.exec_command('configure')
-        if rc != 0:
-            self._module.fail_json(msg='unable to enter configuration mode', output=to_text(err))
+    def get_capabilities(self):
+        connection = self._get_connection()
+        info = connection.get_capabilities()
```

Comment:
We can cache the capabilities information into a dict, rather querying device for every call.